lib/nose/plans/index_lookup.rb

```diff
@@ -279,9 +279,9 @@ def update_state(parent)
 
         # Find fields which are filtered by the index
         @eq_filter = @index.hash_fields + (@state.eq & order_prefix)
-        if order_prefix.include?(@state.range)
-          @range_filter = @state.range
-          @state.range = nil
+        if order_prefix and order_prefix >= Set.new(@state.ranges)
+          @range_filter = @state.ranges
+          @state.ranges = nil
```

Comment:
Isn't it possible that only one of the range conditions could be handled by a particular column family? This case doesn't necessarily need to be handled, but it may be worth considering.